profiles/management/commands/retire_users.py

```diff
@@ -0,0 +1,98 @@
+"""
+Retire user from MM
+"""
+import logging
+from argparse import RawTextHelpFormatter
+from django.contrib.auth.models import User
+from django.core.management import BaseCommand, CommandError
+from social_django.models import UserSocialAuth
+
+from dashboard.models import ProgramEnrollment
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Retire user from MicroMasters
+    """
+    help = """
+Retire one or multiple users. For single user do like this:\n
+`./manage.py retire_users --user=foo` or do \n
+`./manage.py retire_users -u foo` \n
+
+For multiple users, add arg `--user` for each user i.e:\n
+`./manage.py retire_users --user=foo --user=bar --user=baz` or do \n
+`./manage.py retire_users -u foo -u bar -u baz`
+"""
+
+    def create_parser(self, prog_name, subcommand):
+        """
+        create parser to add new line in help text.
+        """
+        parser = super(Command, self).create_parser(prog_name, subcommand)
+        parser.formatter_class = RawTextHelpFormatter
+        return parser
+
+    def add_arguments(self, parser):
+        """create args"""
+        # pylint: disable=expression-not-assigned
+        parser.add_argument(
+            '-u',
+            '--user',
+            action='append',
+            default=[],
+            dest='users',
+            help="Single or multiple user name"
+        ),
+
+    def command_message(self, message, is_error=False):
+        """
+        display and log message
+        """
+        logger = getattr(log, 'error' if is_error else 'info')
+        logger(message)
+        self.stdout.write(message)
+
+    def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
+        user_names = kwargs.get("users", [])
+
+        if len(user_names) <= 0:
+            # show error when no user selected.
+            raise CommandError("Please select user(s)")
+
+        for user_name in user_names:
+            # retire user
+            self.command_message(f"Retiring user {user_name}")
+
+            if not user_name:
+                # invalid user name, can be empty string
+                self.command_message(f"Invalid username: '{user_name}'", True)
+                continue
```

Comment:
Actually, I think you could just omit this test, since if username is null, it's going to fail the next test anyway.

Reply:
i was thinking why hit database if username is null. I think this check will save operation cost.